Corvus.Extensions.Samples/TraversalExtensionsSample.dib

```diff
@@ -0,0 +1,251 @@
+#!markdown
+
+## Traversal Extensions - Sample Notebook
+
+#!markdown
+
+First, add a reference to the `Corvus.Extensions` NuGet package.
+
+#!csharp
+
+#r "nuget: Corvus.Extensions, 1.1.4"
+
+using Corvus.Extensions;
+
+#!markdown
+
+### ForEachAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable.
+
+#!csharp
+
+public async Task<bool> isEqualToOne(int x)
+{
+    bool result = x.Equals(1);
+    Console.WriteLine(result);
+    return result;
+}
+
+Func<int, Task<bool>> funcIsEqualToOne = isEqualToOne;
+
+#!csharp
+
+var list = new List<int> { 1, 2, 3, 4, 5 };
+await list.ForEachAsync<int>(isEqualToOne)
+
+#!markdown
+
+The `ForEachAsync()` method is here for historical interest. It is now possible to write the following and get the same output:
+
+#!csharp
+
+foreach (int x in list)
+{
+    await isEqualToOne(x);
+}
+
+#!markdown
+
+### ForEachAtIndex<T>()
+
+#!markdown
+
+Execute an action for each item in the enumerable with the index of the item in the enumerable.
+
+#!csharp
+
+static void ConsolePrint(string s, int i)
+{
+    Console.WriteLine(i);
+    Console.WriteLine(s);
+}
+
+Action<string, int> printAction = ConsolePrint;
+
+#!csharp
+
+IEnumerable<string> list = new List<string> { "foo", "bar", "quux" };
+list.ForEachAtIndex<string>(printAction);
+
+#!csharp
```

Comment:
I don't understand the purpose of this cell.

Reply:
Not sure what you mean. The cell shows an example of an implementation of `ForEachAtIndex<T>()` on the list declared.

Reply:
So are you saying that the following cell is here to show us what, in effect, `list.ForEachAtIndex` is doing for us?

Maybe add a markdown cell in front of it explaining that this is what it's meant to be showing?

Reply:
Yes, sorry, I was referring to the cell that contains this:

```
int i = 0;

foreach (string v in list)
{
    printAction(v, i++);
}
```

As Ian says, it looks like it's showing us what `ForEachAtIndex` is doing internally, but that's not something we've done anywhere else so I didn't understand why we're doing it here.